src/Core/Scheduler/Providers/3dTiles_Provider.js

```diff
@@ -80,12 +81,39 @@ function getBox(volume) {
     }
 }
 
+const rePosition = new RegExp('gl_Position.*(?![^]*gl_Position)');
+const reMain = new RegExp('[^\\w]*main[^\\w]*(void)?[^\\w]*{');
+function patchMaterial(material) {
+    // Add vertex shader log depth buffer header
+    material.vertexShader = `#include <logdepthbuf_pars_vertex>\n#define EPSILON 1e-6\n${material.vertexShader}`;
+    // Add log depth buffer code snippet after last gl_Position modification
+    let re = rePosition.exec(material.vertexShader);
+    let idx = re[0].length + re.index;
+    material.vertexShader = `${material.vertexShader.slice(0, idx)}\n#include <logdepthbuf_vertex>\n${material.vertexShader.slice(idx)}`;
+
+    // Add fragment shader log depth buffer header
+    material.fragmentShader = `precision highp float;\n#include <logdepthbuf_pars_fragment>\n${material.fragmentShader}`;
+    // Add log depth buffer code snippet at the first line of the main function
+    re = reMain.exec(material.fragmentShader);
+    idx = re[0].length + re.index;
+    material.fragmentShader = `${material.fragmentShader.slice(0, idx)}\n#include <logdepthbuf_fragment>\n${material.fragmentShader.slice(idx)}`;
+
+    material.defines = {
+        USE_LOGDEPTHBUF: 1,
+        USE_LOGDEPTHBUF_EXT: 1,
+    };
+}
+
 $3dTiles_Provider.prototype.b3dmToMesh = function b3dmToMesh(data, layer) {
     return this.b3dmLoader.parse(data).then((result) => {
         const init = function f_init(mesh) {
             mesh.frustumCulled = false;
-            if (layer.overrideMaterials) {
-                mesh.material = new THREE.MeshLambertMaterial(0xffffff);
+            if (mesh.material) {
+                if (layer.overrideMaterials) {
+                    mesh.material = new THREE.MeshLambertMaterial(0xffffff);
+                } else if (Capabilities.isLogDepthBufferSupported()) {
+                    patchMaterial(mesh.material);
```

Comment:
Maybe you could also make sure the shader doesn't already implement this?
(maybe looking for `USE_LOGDEPTHBUF`, `#include <logdepthbuf` strings ? or `gl_FragDepthEXT` writes in the fragment shader?

In all cases I'd suggest also to:
- log a warning indicating that we patched the file
- add a way for the user to disable this patch (an option in layer for instance ?)